HLTrigger/Configuration/python/customizeHLTTrackingForPhaseI2017.py

```diff
@@ -1,89 +1,91 @@
-def customizeHLTForPFTrackingPhaseI2017(process):
+def modifyHLTforPhaseIPFTracking(fragment):
+    """Modify the HLT configuration to run the Phase I tracking in the particle flow sequence"""
 
-	process = customizeHLTPhaseIPixelGeom(process)
-
-	process.hltPixelLayerTriplets.layerList = cms.vstring(
-	    'BPix1+BPix2+BPix3',
-	    'BPix2+BPix3+BPix4',
-	    'BPix1+BPix3+BPix4',
-	    'BPix1+BPix2+BPix4',
-	    'BPix2+BPix3+FPix1_pos',
-	    'BPix2+BPix3+FPix1_neg',
-	    'BPix1+BPix2+FPix1_pos',
-	    'BPix1+BPix2+FPix1_neg',
-	    'BPix2+FPix1_pos+FPix2_pos',
-	    'BPix2+FPix1_neg+FPix2_neg',
-	    'BPix1+FPix1_pos+FPix2_pos',
-	    'BPix1+FPix1_neg+FPix2_neg',
-	    'FPix1_pos+FPix2_pos+FPix3_pos',
-	    'FPix1_neg+FPix2_neg+FPix3_neg' 
-	)
-
-	process.hltPixelLayerQuadruplets = cms.EDProducer("SeedingLayersEDProducer",
-		 BPix = cms.PSet(
-		 	useErrorsFromParam = cms.bool( True ),
-		        hitErrorRPhi = cms.double( 0.0027 ),
-		        TTRHBuilder = cms.string( "hltESPTTRHBuilderPixelOnly" ),
-	                HitProducer = cms.string( "hltSiPixelRecHits" ),
-		        hitErrorRZ = cms.double( 0.006 )
-		    ),
-		FPix = cms.PSet(
-		      useErrorsFromParam = cms.bool( True ),
-		      hitErrorRPhi = cms.double( 0.0051 ),
-		      TTRHBuilder = cms.string( "hltESPTTRHBuilderPixelOnly" ),
-		      HitProducer = cms.string( "hltSiPixelRecHits" ),
-		      hitErrorRZ = cms.double( 0.0036 )
-		    ),
-	    MTEC = cms.PSet( ),
-	    MTIB = cms.PSet( ),
-	    MTID = cms.PSet( ),
-	    MTOB = cms.PSet( ),
-	    TEC = cms.PSet( ),
-	    TIB = cms.PSet( ),
-	    TID = cms.PSet( ),
-	    TOB = cms.PSet( ),
-	    layerList = cms.vstring(
-	        'BPix1+BPix2+BPix3+BPix4', 
-	        'BPix1+BPix2+BPix3+FPix1_pos', 
-	        'BPix1+BPix2+BPix3+FPix1_neg', 
-	        'BPix1+BPix2+FPix1_pos+FPix2_pos', 
-	        'BPix1+BPix2+FPix1_neg+FPix2_neg', 
-	        'BPix1+FPix1_pos+FPix2_pos+FPix3_pos', 
-	        'BPix1+FPix1_neg+FPix2_neg+FPix3_neg'
-	    )
-	)
-
-
-   
+    # hltPixelLayerTriplets
+    if hasattr(fragment, "hltPixelLayerTriplets"):
+        phase1Pixel.toModify( fragment.hltPixelLayerTriplets,
+            layerList = cms.vstring(
+                'BPix1+BPix2+BPix3',
+                'BPix2+BPix3+BPix4',
+                'BPix1+BPix3+BPix4',
+                'BPix1+BPix2+BPix4',
+                'BPix2+BPix3+FPix1_pos',
+                'BPix2+BPix3+FPix1_neg',
+                'BPix1+BPix2+FPix1_pos',
+                'BPix1+BPix2+FPix1_neg',
+                'BPix2+FPix1_pos+FPix2_pos',
+                'BPix2+FPix1_neg+FPix2_neg',
+                'BPix1+FPix1_pos+FPix2_pos',
+                'BPix1+FPix1_neg+FPix2_neg',
+                'FPix1_pos+FPix2_pos+FPix3_pos',
+                'FPix1_neg+FPix2_neg+FPix3_neg'
+            )
+        )
 
+    # hltPixelLayerQuadruplets
+    phase1Pixel.toModify( fragment, lambda fragment:
+        setattr(fragment, "hltPixelLayerQuadruplets", cms.EDProducer("SeedingLayersEDProducer",
```

Comment:
@fwyzard Since you're passing the full `fragment` to `cms.Modifier`, I wonder if 
```
phase1Pixel.toModify(fragment, hltPixelLayerQuadruplets = cms.EDProducer("...", ...)
```
might work instead?

(maybe you tried it already and it didn't work; in fact I'd guess the answer would be "no" because IIRC `toModify()` needs similar support as `clone()` from the object, and I'm not sure if `fragment` has such)